### ISSUE TITLE
Compiler error when compiling SU2_AD with intel compilers

### ISSUE THREAD

**Issue by jayantmukho** [su2code/SU2#719]
I am trying to compile SU2 for parallel runs using intel compilers. SU2 compiles perfectly when compiled without the --enable-autodiff flag, but throws an error when compiling with it. The error is: 

```
../../../Common/lib/../src/adt_structure.cpp(1402): internal error: assertion failed at: “shared/cfe/edgcpfe/exprutil.c”, line 7266

   const su2double V0x = xc[0] - 0.5*xRel[4][0] - 0.125*(xRel[1][0]+xRel[2][0]+xRel[3][0]);
                                     ^

compilation aborted for ../../../Common/lib/../src/adt_structure.cpp (code 4)
make[2]: *** [../src/libSU2_AD_a-adt_structure.o] Error 4
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/path/SU2_AD/Common/lib’
make[1]: *** [install-recursive] Error 1
make[1]: Leaving directory `/path/SU2_AD’
make: *** [install-SU2_AD] Error 2
```

Looking for the error online, it seems to be something related to the compiler itself. I was wondering if anyone else had encountered this error message, and how they got around it. 

I am using icc and icpc version 18.0.1

**Comment by WallyMaier** [comment]
This used to work on 18.0.1, but I am getting the same issue.  v19.0.4 works, however.

**Comment by jayantmukho** [comment]
I can't confirm that using Intel v19.0.4 worked (since I was unable to get that compiler version) but I am closing the issue nonetheless since @WallyMaier confirmed that he got the same error and fixed it by using v19.0.4
